### ISSUE TITLE
Support WASIp2 sockets in `read` and `write`

### ISSUE THREAD

**Issue by dicej** [WebAssembly/wasi-libc#521]
Currently, you can use `recv` and `send` to read from and write to WASIp2 sockets, respectively, but neither `read` nor `write` work.  We should support those functions as well, using logic analogous to what we have in `close`, where we check whether the file descriptor is a WASIp2 socket and defer to the appropriate function if it is: https://github.com/WebAssembly/wasi-libc/blob/b9ef79d7dbd47c6c5bafdae760823467c2f60b70/libc-bottom-half/sources/__wasilibc_fd_renumber.c#L78-L95

(Note that whereas we use `descriptor_table_remove` in the `close` implementation, we'll want to use `descriptor_table_get_ref` in the `read` and `write` implementations.)

**Comment by abrown** [comment]
@dicej, is this still an issue?

**Comment by dicej** [comment]
> is this still an issue?

Yes; neither [read](https://github.com/WebAssembly/wasi-libc/blob/main/libc-bottom-half/cloudlibc/src/libc/unistd/read.c) nor [write](https://github.com/WebAssembly/wasi-libc/blob/main/libc-bottom-half/cloudlibc/src/libc/unistd/write.c) have WASIp2 socket support yet.  Shouldn't be hard to implement, but nobody has gotten around to it yet.
